spec/enumerables_spec.rb

```diff
@@ -0,0 +1,129 @@
+require_relative '../enumerables.rb'
+
+describe Enumerable do
+  let(:array) { [1, 2, 3, 4, 5, 6, 7, 8, 9] }
+  let(:words_array) { %w[ant bear cat] }
+  let(:nil_true) { [nil, true, 99, false] }
+  let(:proced) { proc { |n| n * 2 } }
+  describe '#my_each' do
+    it 'Returns each element' do
+      expect(array.my_each { |n| n * n }).to eql(array)
+    end
+
+    it 'Returns enumerator' do
+      expect(array.my_each.class).to be(Enumerator)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'Returns each element' do
+      expect(array.my_each_with_index { |n| n * n }).to eql(array)
+    end
+
+    it 'Returns enumerator' do
+      expect(array.my_each_with_index.class).to be(Enumerator)
+    end
+  end
+
+  describe '#my_select' do
+    it 'Returns even numbers' do
+      expect(array.select(&:even?)).to eql([2, 4, 6, 8])
+    end
+
+    it 'Returns words shorter than 4 characters' do
+      expect(words_array.select { |n| n.length < 4 }).to eql(%w[ant cat])
+    end
+
+    it 'Returns enumerator' do
+      expect(array.my_select.class).to be(Enumerator)
+    end
+  end
+
+  describe '#my_all?' do
+    it 'Returns false for even numbers' do
+      expect(array.my_all?(&:even?)).to eql(false)
+    end
+
+    it 'Return true for words smaller than 5' do
+      expect(words_array.my_all? { |n| n.length < 5 }).to eql(true)
+    end
+
+    it 'Return false for all equal' do
+      expect(nil_true.my_all?).to eql(false)
+    end
+  end
+
+  describe '#my_any?' do
+    it 'Returns true if any number is even' do
+      expect(array.my_any?(&:even?)).to eql(true)
+    end
+
+    it 'Returns false for words shorter than 3 characters' do
+      expect(words_array.my_any? { |n| n.length < 3 }).to eql(false)
+    end
+
+    it 'Returns true for any element in an array' do
+      expect(nil_true.my_any?).to eql(true)
+    end
+  end
```

Comment:
Add the following test cases to make it complete

When a class is given as a parameter
When a regex is given
When a value is given as a parameter

Reply:
Done! Thank you for pointing it out! We added all the corner case tests for the method!